kasa/protocol.py

```diff
@@ -156,7 +156,7 @@ async def _query(self, request: str, retry_count: int, timeout: int) -> Dict:
 
     def __del__(self) -> None:
         if self.writer and self.loop and self.loop.is_running():
-            self.writer.close()
+            self.loop.call_soon_threadsafe(self.writer.close)
```

Comment:
Mind adding a quick note about why this is necessary, for any future readers?

Reply:
done